WPFHospitalEditor/Service/MapObjectService.cs

```diff
@@ -38,5 +45,29 @@ private List<MapObject> colorSelectedObject(List<MapObject> allMapObjects)
             }
             return allMapObjects;
         }
+
+        public List<MapObject> searchForMapObjects(string name, string type)
+        {
+            var mapObjects = new List<MapObject>();
+            List<MapObject> allMapObjects = iMapObjectRepository.getAllMapObjects();
+            if (name.Equals("") && type.Equals(AllConstants.emptyComboBox)) return mapObjects;
```

Comment:
Just switch these two lines because if it's empty input you will call repo without reason.